Userland/Libraries/LibWeb/SVG/SVGScriptElement.cpp

```diff
@@ -19,4 +20,41 @@ void SVGScriptElement::initialize(JS::Realm& realm)
     set_prototype(&Bindings::ensure_web_prototype<Bindings::SVGScriptElementPrototype>(realm, "SVGScriptElement"));
 }
 
+void SVGScriptElement::visit_edges(Cell::Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    visitor.visit(m_script);
+}
+
+// https://www.w3.org/TR/SVGMobile12/script.html#ScriptContentProcessing
+void SVGScriptElement::process_the_script_element()
+{
+    // 1. If the 'script' element's "already processed" flag is true or if the element is not in the
+    //    document tree, then no action is performed and these steps are ended.
+    if (m_already_processed || !in_a_document_tree())
+        return;
+
+    // FIXME: 2. If the 'script' element references external script content, then the external script content
+    //           using the current value of the 'xlink:href' attribute is fetched. Further processing of the
+    //           'script' element is dependent on the external script content, and will block here until the
+    //           resource has been fetched or is determined to be an invalid IRI reference.
+    if (has_attribute(HTML::AttributeNames::href)) {
```

Comment:
To check the `xlink:href` attribute I think we need to instead do something like:

```suggestion
    if (has_attribute_ns("href", Namespaces::XLink)) {
```

`xlink:href` is not the same as `href`. (I don't know if it's OK to use `HTML::AttributeNames::href`, we don't have an XLink equivalent I think? Adding `XLink::AttributeNames` somewhere would probably be the best.)

Reply:
Hmm, interesting - yeah that doesn't seem to be working. I guess something may be wrong with the parser here then? I can't see the attribute created with that namespace.

Will investigate some more :^)

Reply:
Looks like we're creating the script element in the SVG namespace here: https://github.com/SerenityOS/serenity/blob/5e15c29e22cb4dfab8889bba59af63a033a8457b/Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp#L3511

Even though we adjust the current token to be in the XLink namespace in the step before.

🤔

Reply:
Okay - I think I may have solved the issue I mentioned above. Although I think I may need to check both the namespaced and non-namespaced href attribute if I am understanding correctly?